### PR TITLE
Add doc for multi-file hive buckets support

### PR DESCRIPTION
**Pull request by losipiuk**

### CODE REVIEW

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 1]
```diff
@@ -284,6 +287,27 @@ This table can then be queried in Presto::
...
 
+.. _clustered-tables:
+
+Clustered hive tables support
+-----------------------------
+
+By default presto supports only one data file per bucket per partition for clustered tables (Hive tables declared with ``CLUSTERED BY`` clause).
+If number of files does not match number of buckets exception would be thrown.
+
+The enable support for cases when multiple INSERTs where done to single partition of clustered user may use:
```
Comment (anusudarsan) [inline presto-docs/src/main/sphinx/connector/hive.rst]: Maybe rewrite this line to "To enable support for cases where there are more than one file per bucket, when multiple INSERTs are done to a single partition of the clustered table, you can use:"

File: presto-docs/src/main/sphinx/connector/hive.rst [hunk 2]
```diff
@@ -284,6 +287,27 @@ This table can then be queried in Presto::
...
+
+ * ``hive.multi-file-bucketing.enabled`` config property
+ * ``multi_file_bucketing_enabled`` session property (using ``SET SESSION <connector_name>.multi_file_bucketing_enabled``)
+
+Config property changes behaviour globally and session property can be used on per query basis.
+The default value of session property is taken from config property.
+
+If support for multiple files per bucket is enabled Presto will group the files in partition directory.
+It will sort filenames lexicographically. Then it will treat part of filename up to first underscore character as bucket key.
```
Comment (anusudarsan) [inline presto-docs/src/main/sphinx/connector/hive.rst:307]: maybe an example will help?

**Comment by anusudarsan** [comment]
LGTM mod comments

**Comment by mattsfuller** [comment]
I can address comments and merge the change.

**Pull request closed without merging**